src/cpu/x64/asm.rs

```diff
@@ -1057,6 +1057,10 @@ pub fn cvttsd2si(buf: &mut MacroAssembler, x64: u8, dest: Reg, src: FReg) {
     sse_float_reg_freg(buf, true, 0x2c, x64, dest, src);
 }
 
+pub fn movd(buf: &mut MacroAssembler, dest: FReg, x64: u8, src: Reg) {
```

Comment:
I usually add tests for assembly instructions at the bottom of the file. Please add a few test cases with different registers.  You can use [this online assembler](https://defuse.ca/online-x86-assembler.htm) to get the encoded bytes for an assembly instruction.

Reply:
I've run your PR locally and the failure is due to a SIGILL (an illegal instruction), so it seems to me that the instruction encoding is not right. Writing tests with `assert_emit!` combined with the online assembler should tell you what's wrong - not too much assembly required just run the unit tests until the instruction encodings match. If you need more help or don't want to look into this just ping me again.

Reply:
Thanks for the link, I'll have a look and try things the way you recommend. I'll report back.

Reply:
Your link was really helpful, thanks!